cpp/src/arrow/meson.build

```diff
@@ -289,6 +373,49 @@ install_headers(
     install_dir: 'arrow',
 )
 
+if needs_tests
+    filesystem_dep = dependency('boost', modules: ['filesystem'])
+    gtest_main_dep = dependency('gtest_main')
+    gmock_dep = dependency('gmock')
+else
+    filesystem_dep = disabler()
+    gtest_main_dep = disabler()
+    gmock_dep = disabler()
+endif
+
+arrow_test_lib = static_library(
```

Comment:
The static_library is required here because the Meson wrap dependency for gtest only provides consumers with the source files, and when using that in two different shared libraries then linking together you get an ODR violation.

See also https://github.com/mesonbuild/wrapdb/issues/1937